homeassistant/components/mqtt/__init__.py

```diff
@@ -46,15 +46,17 @@
 ATTR_TOPIC = 'topic'
 ATTR_PAYLOAD = 'payload'
 ATTR_QOS = 'qos'
+ATTR_RETAIN = 'retain'
 
 MAX_RECONNECT_WAIT = 300  # seconds
 
 
-def publish(hass, topic, payload, qos=None):
+def publish(hass, topic, payload, qos=None, retain=False):
```

Comment:
This publish method is a helper method to call the actual service. It is not responsible for setting defaults. That is done by the service handler which is defined within the setup method.